src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -105,8 +115,46 @@ protected internal override PinMode GetPinMode(int pinNumber)
 
         protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
         {
-            // Libgpiod Api do not support pull up or pull down resistors for now.
-            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+            switch (mode)
+            {
+                case PinMode.Input:
+                    return true;
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                    // for use the bias flags we need libgpiod version 1.5 or later
+                    IntPtr libgpiodVersionPtr = Interop.libgpiod.gpiod_version_string();
+                    string libgpiodVersionMatch = Marshal.PtrToStringAnsi(libgpiodVersionPtr);
+                    Version libgpiodVersion = new Version(libgpiodVersionMatch);
+                    bool isLibgpiod1dot5 = (libgpiodVersion.Major >= 1 && libgpiodVersion.Minor >= 5);
+
+                    // for use the bias flags we need Kernel version 5.5 or later
+                    string[] linuxVersionMatch = RuntimeInformation.OSDescription.Replace("Linux ", string.Empty).Split('.');
+                    Version linuxVersion = new Version();
+                    bool isKernelLinux5dot5 = false;
+
+                    if (linuxVersionMatch.Length > 1)
+                    {
+                        linuxVersion = new Version($"{linuxVersionMatch[0]}.{linuxVersionMatch[1]}");
+                        isKernelLinux5dot5 = (linuxVersion.Major >= 5 && libgpiodVersion.Minor >= 5);
+                    }
+
+                    // check if we have the correct versions
+                    if (isKernelLinux5dot5 && isLibgpiod1dot5)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        Console.Error.WriteLine($"Using Kernel v{linuxVersion} but v5.5 or later is required.\n" +
```

Comment:
I don't believe `IsPinModeSupported` should be printing anything since it's used for probing. You can move this logic to separate method and then whenever this mode is trying to be used only then improve current error message.

Also the linux kernel version can be checked only once:
create a static method for probing: `IsLinuxKernelVersion1_5OrHigher` and then assign this to a static field. Note this will be lazy initialized on the first usage of this class.